pkg/controller/endpointslice/reconciler.go

```diff
@@ -262,14 +266,23 @@ func (r *reconciler) reconcileByAddressType(service *corev1.Service, pods []*cor
 	}
 
 	if r.topologyCache != nil && hintsEnabled(service.Annotations) {
-		slicesToCreate, slicesToUpdate = r.topologyCache.AddHints(si)
+		slicesToCreate, slicesToUpdate, events = r.topologyCache.AddHints(si)
 	} else {
 		if r.topologyCache != nil {
+			if r.topologyCache.HasPopulatedHints(si.ServiceKey) {
+				events = append(events, &topologycache.EventBuilder{
+					EventType: corev1.EventTypeWarning,
+					Reason:    "TopologyAwareHintsDisabled",
+					Message:   fmt.Sprintf("Topology Aware Hints has been disabled, removing %s hints from service %q", addressType, si.ServiceKey),
+				})
+			}
 			r.topologyCache.RemoveHints(si.ServiceKey, addressType)
 		}
 		slicesToCreate, slicesToUpdate = topologycache.RemoveHintsFromSlices(si)
 	}
-
+	for _, event := range events {
+		r.eventRecorder.Event(service, event.EventType, event.Reason, event.Message)
+	}
```

Comment:
@robscott Should I move it after finalize()?

Reply:
I think these events are describing something that has happened, not something that will, so in that sense, I'd prefer to put it after the finalize call.